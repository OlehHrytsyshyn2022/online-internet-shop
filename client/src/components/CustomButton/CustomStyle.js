export const customButtonStyle = {
  background: "black",
  height: "53px",
  width: "30vw",
  maxWidth: "500px",
};
