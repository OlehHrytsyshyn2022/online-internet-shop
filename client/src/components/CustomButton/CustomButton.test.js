import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import CustomButton from "./CustomButton";

describe("CustomButton", () => {
  test("renders CustomButton component", () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <CustomButton value="Button" onClick={onClick} />
    );
    const CustomButtonText = getByText("Button");
    expect(CustomButtonText).toBeInTheDocument();
  });

  test("Snapshot Test CustomButton", () => {
    const { container } = render(<CustomButton />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onClick handler when clicked", () => {
    const onClickMock = jest.fn();
    render(<CustomButton value="Click me" onClick={onClickMock} />);
    fireEvent.click(screen.getByText("Click me"));
    expect(onClickMock).toHaveBeenCalled();
  });
});
