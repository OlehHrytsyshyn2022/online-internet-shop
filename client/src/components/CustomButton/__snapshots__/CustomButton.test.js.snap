// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Snapshot Test CustomButton 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
  style="background: black; height: 53px; width: 30vw; max-width: 500px;"
  tabindex="0"
  type="button"
>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
